Yoast/wordpress-seo
# Dupp 328 organization add tailwind input boxes for socials

## Context


* For organizations, we want  to allow the user to specify a variable number of social profiles URLS instead of  presenting a fixed set of choices. 

## Summary


This PR can be summarized in the following changelog entry:

* Change the social profiles step for organizations in order to accomodate for a variable number of text fields representing social profiles URLS.

## Relevant technical choices:

* The component managing the variable number of text inputs is based on [Admin-UI FieldArray component ](https://github.com/Yoast/admin-ui/blob/trunk/packages/settings/src/components/field-array.js)
* This PR does not include variable text inputs validation

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Navigate to the first-time configuration social profiles step
* Open the javascript console in your browser
* In the console, write: ` window.wpseoWorkoutsData.configuration.socialProfiles.otherSocialUrls` and verify the _OtherSocialUrls_ array is empty
* Click on `Add another URL` and write something in the text field that appears
* Repeat the step above two times
* Click on the `Save and continue` button
* Refresh the page and write again ` window.wpseoWorkoutsData.configuration.socialProfiles.otherSocialUrls` in the console
* Verify that now _OtherSocialUrls_ array contains the value specified in the three text fields added
* Click on the trashcan button aside the second newly-created text field, in order to delete it
* Refresh the page and write again ` window.wpseoWorkoutsData.configuration.socialProfiles.otherSocialUrls` in the console
* Verify that now _OtherSocialUrls_ array contains the values specified in the two remaining text fields

### Test instructions for QA when the code is in the RC


* **Please be aware that when you delete a text field other than the last one added, the ids related to the remaining text fields and associated  trashcan buttons  are automatically rearranged to maintain sequential numbering without gaps.**
* This means that if you have three text fields with ids: `social-input-other-url-1`, `social-input-other-url-2`, and `social-input-other-url-3` respectively, if you delete the one with the id `social-input-other-url-2`, the two remaining text fields will have the ids `social-input-other-url-1` and `social-input-other-url-2`,  not `social-input-other-url-1` and `social-input-other-url-3` as expected

QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes [DUPP-328](https://yoast.atlassian.net/browse/DUPP-328)
